nixos/modules/security/apparmor.nix

```diff
@@ -29,6 +39,577 @@ in
+     environment.etc."/apparmor.d/abstractions".source = pkgs.apparmor-profiles + "/etc/apparmor.d/abstractions";
+     environment.etc."/apparmor.d/tunables".source = pkgs.apparmor-profiles + "/etc/apparmor.d/tunables";
+
+     environment.etc."apparmor/parser.conf".text = ''
+      Optimize=compress-fast
+      ${cfg.extraParserConfig}
+    '';
+
+     # Pretty much everything in here, is there to
+     # prevent apparmor utilities from crashing.
+     environment.etc."apparmor/logprof.conf".text =
+      builtins.replaceStrings
+      [
+        "/usr"
+      ]
+      [
+        "/nix/store/*"
+      ]
+     ''
+      [settings]
+      profiledir = /etc/apparmor.d
+      inactive_profiledir = /usr/share/apparmor/extra-profiles 
+      parser = ${pkgs.apparmor-parser}/bin/apparmor_parser
+      logfiles = /var/log/messages 
+
+      ldd = ${pkgs.glibc.bin}/bin/ldd
+      logger = ${pkgs.logger}/bin/logger
+      default_owner_prompt = 1
+      custom_includes = 
+
+      [qualifiers]
+
+      [globs]
+      # /foo/bar/lib/libbaz.so -> /foo/bar/lib/lib*
+      /lib/lib[^\/]+so[^\/]*$           = /lib/lib*so*
+
+      # strip kernel version numbers from kernel module accesses
+      ^/lib/modules/[^\/]+\/            = /lib/modules/*/
+
+      # strip pid numbers from /proc accesses
+      ^/proc/\d+/                       = /proc/*/
+
+      # if it looks like a home directory, glob out the username
+      ^/home/[^\/]+                     = /home/*
+
+      # if they use any perl modules, grant access to all
+      ^/usr/lib/perl5/.+$               = /usr/lib/perl5/**
+      ^/usr/lib/[^\/]+/perl5?/.+$       = /usr/lib/@{multiarch}/perl{,5}/**
+
+      # locale foo
+      ^/usr/lib/locale/.+$              = /usr/lib/locale/**
+      ^/usr/share/locale/.+$            = /usr/share/locale/**
+
+      # timezone fun
+      ^/usr/share/zoneinfo/.+$          = /usr/share/zoneinfo/**
+
+      # /foobar/fonts/baz -> /foobar/fonts/**
+      /fonts/.+$                        = /fonts/**
+
+      # turn /foo/bar/baz.8907234 into /foo/bar/baz.*
+      # BUGBUG - this one looked weird because it would suggest a glob for
+      # BUGBUG - libfoo.so.5.6.0 that looks like libfoo.so.5.6.*
+      # \.\d+$                            = .*
+
+      # some various /etc/security poo -- dunno about these ones...
+      ^/etc/security/_[^\/]+$           = /etc/security/*
+      ^/lib/security/pam_filter/[^\/]+$ = /lib/security/pam_filter/*
+      ^/lib/security/pam_[^\/]+\.so$    = /lib/security/pam_*.so
+
+      ^/etc/pam.d/[^\/]+$               = /etc/pam.d/*
+      ^/etc/profile.d/[^\/]+\.sh$       = /etc/profile.d/*.sh
+
+      [required_hats]
+
+      [defaulthat]
+
+      [repository]
+        distro         = ubuntu-intrepid
+        url            = http://apparmor.test.opensuse.org/backend/api
+        preferred_user = ubuntu
+
+      ${cfg.extraLogConfig}
+     '';
+
+     environment.etc."/apparmor/severity.db".text =
```

Comment:
Same here, can we take this from apparmor directly?

Reply:
Yes but no but yes, see above ^^